### PR TITLE
CMake corrections

### PR DESCRIPTION
**Pull request by mpuchner**
This PR addresses mainly the issue 2511, "CMake: FindCerf.cmake behaves inconsistent" 

see http://apps.jcns.fz-juelich.de/redmine/issues/2511

in addition a few mini-corrections

### CODE REVIEW

File: cmake/find/FindCerf.cmake [hunk 1]
```diff
@@ -32,32 +32,57 @@ find_package_handle_standard_args(Cerf
...
-        message(STATUS "Found neither libcerf nor libcerfcpp")
-        return()
-    endif()
-    message(STATUS "Found libcerf, language=CPP, version=${Cerf_VERSION}, lib=${Cerf_LIBRARIES},"
+		
+    if(Cerf_FOUND)
+		set(Cerf_IS_CPP ON CACHE BOOL "Define whether CERF shall be used as C++")
+	endif()
+endif()
+
```
Comment (jwuttke) [inline cmake/find/FindCerf.cmake:59]: no double or triple blank lines, and no blank lines in the if block below, for consistence of style please

File: cmake/find/FindCerf.cmake [hunk 2]
```diff
@@ -32,32 +32,57 @@ find_package_handle_standard_args(Cerf
...
+
+# If still not found, then check whether it is ultimately required or not.
+if(NOT Cerf_FOUND)
+	message(STATUS "libcerf: FOUND=${Cerf_FOUND}, VERSION=${Cerf_VERSION}, "
+		"LIB=${Cerf_LIBRARIES}, IS_CPP=${Cerf_IS_CPP}")
+		
+	if(${Cerf_ULTIMATELY_REQUIRED})
+		message(FATAL_ERROR "Found neither C nor C++ version of cerf")
+	else()
```
Comment (jwuttke) [inline cmake/find/FindCerf.cmake:68]: no else, as we know that execution terminates after FATAL_ERROR

**Review by jwuttke — CHANGES_REQUESTED** [review]
Thank you for sorting out this dependency hell.

File: cmake/find/FindCerf.cmake (hunk 1 above)
Comment (mpuchner) [reply cmake/find/FindCerf.cmake:59]: Thanks for the hint, didn't know this. Will remove them

File: cmake/find/FindCerf.cmake (hunk 2 above)
Comment (mpuchner) [reply cmake/find/FindCerf.cmake]: wiel change it

**Pull request merged**